### PR TITLE
DDFFORM 900 material warning backend

### PR DESCRIPTION
**Pull request by LasseStaus**
- **Add error components for material search.**
- **Add css for material search error components.**

#### Link to issue

[DDFFORM-900](https://reload.atlassian.net/browse/DDFFORM-900)

[React pr](https://github.com/danskernesdigitalebibliotek/dpl-react/pull/1304)
[CMS PR](https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/1310)
#### Description

This PR introduceres error/warning components to the material search component. 
They are used to display errors in retrieving the material that was saved before. 

#### Screenshot of the result

image

image

[DDFFORM-900]: https://reload.atlassian.net/browse/DDFFORM-900?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/stories/Library/material-search/Errors/MaterialSearchBaseError.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+const MaterialSearchBaseError: React.FC<MaterialSearchBaseErrorProps> = ({
+  children,
+}) => {
+  return (
+    <div className="material-search__error">
+      <div className="material-search__error-header">
+        <WarningIcon className="material-search__error-icon" />
+        <div>This material is outdated.</div>
```
Comment (kasperg) [inline src/stories/Library/material-search/Errors/MaterialSearchBaseError.tsx:15]: Consider marking this with an actual heading level to signal the priority over the following content/description.

File: src/stories/Library/material-search/Errors/MaterialTypeNotFoundError.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    <MaterialSearchBaseError>
+      <div className="material-search__error-content">
+        <p className="material-search__error-description">
+          The currently selected type of the material is no longer available in
+          the system. As a result of this, the link is likely broken. Use the
+          title or link underneath to find and update the material and its type.
+        </p>
+        <div className="material-search__error-material-content">
+          <div className="material-search__error-item">
+            <span className="material-search__error-term">Title:</span>
```
Comment (kasperg) [inline src/stories/Library/material-search/Errors/MaterialTypeNotFoundError.tsx:21]: Oh - how I still dream of `<dl><dt><dd>` 😅 .

File: src/stories/Library/material-search/MaterialSearch.stories.tsx [hunk 3]
```diff
@@ -5,12 +5,29 @@ import MaterialSearch from "./MaterialSearch";
...
+      control: {
+        type: "boolean",
+      },
+    },
+    displayMaterialTypeError: {
+      defaultValue: false,
+      control: {
+        type: "boolean",
+      },
+    },
```
Comment (kasperg) [inline src/stories/Library/material-search/MaterialSearch.stories.tsx:20]: Consider making this an enum or the like with 3 states - no error, work error, material type error that can be toggled in a single control.

File: src/stories/Library/material-search/material-search.scss [hunk 4]
```diff
@@ -13,6 +13,9 @@ $_box-shadow: 0px 0px 0px 2px rgba(0, 0, 0, 0.25);
 $_loading-spinner-size: 40px;
 $_highlighted-item-bg-color: $_gin-header-bg-color;
 
+$_box_shadow_header: 0 4px 2px -2px rgba(0, 0, 0, 0.171);
+$_alert_background_header: #ecc8c577;
```
Comment (kasperg) [inline src/stories/Library/material-search/material-search.scss:17]: Consider using [red/danger colors from Gin](https://git.drupalcode.org/project/gin/-/blob/8.x-3.x/styles/theme/variables.scss?ref_type=heads#L19-23) to make the element feel more native.

You have used a similar approach elsewhere. How about doing the same here.

**Review by kasperg — APPROVED** [review]

File: src/stories/Library/material-search/MaterialSearch.stories.tsx (hunk 3 above)
Comment (LasseStaus) [reply src/stories/Library/material-search/MaterialSearch.stories.tsx]: Yeah thats cleaner

File: src/stories/Library/material-search/material-search.scss (hunk 4 above)
Comment (LasseStaus) [reply src/stories/Library/material-search/material-search.scss]: Yea agreed. Have changed it to use some gin colors.

**Pull request merged**